### ISSUE TITLE
Slow field editing pages in CP

### ISSUE THREAD

**Issue by evanwarner** [engram-design/SuperTable#346]
After updating a site past Craft 3.4 (and updating SuperTable accordingly), the page load times when accessing a field edit page in the CP have spiked to 40+ seconds. Disabling SuperTable resolves the load times back to normal. I cleared my log files and  tracked a single field edit page load. Here are the lines surrounding the most notable timestamp jumps throughout the log:

```
2020-03-13 13:53:23 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][info][yii\db\Command::execute] INSERT INTO `resourcepaths` (`path`, `hash`) VALUES ('@verbb/supertable/resources/dist', '561fabb6') ON DUPLICATE KEY UPDATE `path`='@verbb/supertable/resources/dist'
2020-03-13 13:53:23 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile begin][yii\db\Command::execute] INSERT INTO `resourcepaths` (`path`, `hash`) VALUES ('@verbb/supertable/resources/dist', '561fabb6') ON DUPLICATE KEY UPDATE `path`='@verbb/supertable/resources/dist'
2020-03-13 13:53:23 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile end][yii\db\Command::execute] INSERT INTO `resourcepaths` (`path`, `hash`) VALUES ('@verbb/supertable/resources/dist', '561fabb6') ON DUPLICATE KEY UPDATE `path`='@verbb/supertable/resources/dist'
2020-03-13 13:53:23 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][info][yii\db\Command::execute] INSERT INTO `resourcepaths` (`path`, `hash`) VALUES ('@verbb/base/resources/dist', 'a6d8719e') ON DUPLICATE KEY UPDATE `path`='@verbb/base/resources/dist'
2020-03-13 13:53:23 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile begin][yii\db\Command::execute] INSERT INTO `resourcepaths` (`path`, `hash`) VALUES ('@verbb/base/resources/dist', 'a6d8719e') ON DUPLICATE KEY UPDATE `path`='@verbb/base/resources/dist'
2020-03-13 13:53:23 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile end][yii\db\Command::execute] INSERT INTO `resourcepaths` (`path`, `hash`) VALUES ('@verbb/base/resources/dist', 'a6d8719e') ON DUPLICATE KEY UPDATE `path`='@verbb/base/resources/dist'
----- 4s delay -----
2020-03-13 13:53:27 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][trace][craft\web\View::renderTemplate] Rendering template: _components/fieldtypes/Matrix/settings
2020-03-13 13:53:27 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile begin][Twig template] render template: _components/fieldtypes/Matrix/settings
2020-03-13 13:53:27 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile begin][Twig template] render macro: field
2020-03-13 13:53:27 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile begin][Twig template] render template: _includes/forms/field
2020-03-13 13:53:27 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile begin][Twig template] render block: input
```

```
2020-03-13 13:53:28 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile begin][Twig template] render template: _includes/forms/errorList
2020-03-13 13:53:28 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile end][Twig template] render template: _includes/forms/errorList
2020-03-13 13:53:28 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile end][Twig template] render template: _includes/forms/field
2020-03-13 13:53:28 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile end][Twig template] render macro: field
2020-03-13 13:53:28 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile end][Twig template] render template: seo/_seo/settings
----- 6s delay -----
2020-03-13 13:53:36 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][trace][craft\web\View::renderTemplate] Rendering template: super-table/settings
2020-03-13 13:53:36 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile begin][Twig template] render template: super-table/settings
2020-03-13 13:53:36 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile begin][Twig template] render macro: selectField
2020-03-13 13:53:36 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile begin][Twig template] render macro: select
2020-03-13 13:53:36 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile begin][Twig template] render template: _includes/forms/select
```

```
2020-03-13 13:53:36 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][info][yii\db\Command::execute] INSERT INTO `resourcepaths` (`path`, `hash`) VALUES ('@app/web/assets/matrixsettings/dist', 'ab617b27') ON DUPLICATE KEY UPDATE `path`='@app/web/assets/matrixsettings/dist'
2020-03-13 13:53:36 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile begin][yii\db\Command::execute] INSERT INTO `resourcepaths` (`path`, `hash`) VALUES ('@app/web/assets/matrixsettings/dist', 'ab617b27') ON DUPLICATE KEY UPDATE `path`='@app/web/assets/matrixsettings/dist'
2020-03-13 13:53:36 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile end][yii\db\Command::execute] INSERT INTO `resourcepaths` (`path`, `hash`) VALUES ('@app/web/assets/matrixsettings/dist', 'ab617b27') ON DUPLICATE KEY UPDATE `path`='@app/web/assets/matrixsettings/dist'
----- 11s delay -----
2020-03-13 13:53:47 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][trace][craft\web\View::renderTemplate] Rendering template: _components/fieldtypes/Matrix/settings
2020-03-13 13:53:47 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile begin][Twig template] render template: _components/fieldtypes/Matrix/settings
2020-03-13 13:53:47 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile begin][Twig template] render macro: field
2020-03-13 13:53:47 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile begin][Twig template] render template: _includes/forms/field
2020-03-13 13:53:47 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile begin][Twig template] render block: input
```

```
2020-03-13 13:53:49 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile end][Twig template] render template: _includes/forms/errorList
2020-03-13 13:53:49 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile end][Twig template] render template: _includes/forms/field
2020-03-13 13:53:49 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile end][Twig template] render macro: field
2020-03-13 13:53:49 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile end][Twig template] render template: seo/_seo/settings
----- 4s delay -----
2020-03-13 13:53:53 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][trace][craft\web\View::renderTemplate] Rendering template: _components/fieldtypes/Matrix/settings
2020-03-13 13:53:53 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile begin][Twig template] render template: _components/fieldtypes/Matrix/settings
2020-03-13 13:53:53 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile begin][Twig template] render macro: field
2020-03-13 13:53:53 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile begin][Twig template] render template: _includes/forms/field
2020-03-13 13:53:53 [-][1][utvh3sa7nvs55i1f1ppqrn6jp7][profile begin][Twig template] render block: input
```

```
2020-03-13 13:53:53 [-][1][-][profile end][Twig template] render template: _includes/forms/errorList
2020-03-13 13:53:53 [-][1][-][profile end][Twig template] render template: _includes/forms/field
2020-03-13 13:53:53 [-][1][-][profile end][Twig template] render macro: field
2020-03-13 13:53:53 [-][1][-][profile end][Twig template] render template: seo/_seo/settings
----- 7s delay -----
2020-03-13 13:54:00 [-][1][-][trace][craft\web\View::renderTemplate] Rendering template: super-table/settings
2020-03-13 13:54:00 [-][1][-][profile begin][Twig template] render template: super-table/settings
2020-03-13 13:54:00 [-][1][-][profile begin][Twig template] render macro: selectField
2020-03-13 13:54:00 [-][1][-][profile begin][Twig template] render macro: select
2020-03-13 13:54:00 [-][1][-][profile begin][Twig template] render template: _includes/forms/select
```

### Additional info

- Plugin version: 2.4.6
- Craft version: 3.4.9

**Comment by tim-van-dijkhuizen** [comment]
@engram-design Sorry for the mention but this issue is pretty important.

My entry index (and most other pages) loads in about 600ms while my entry edit page (with SuperTable) takes more than 14 seconds to load.

I hope this issue gets resolved soon. Thanks in advance!

**Comment by engram-design** [comment]
I'll see what can be done on the Super Table level here. The issue I believe is that when loading the settings for a field, we're pre-loading all fields' HTML/JS. Will look to see if there's any way to mitigate this without stepping into Craft itself's territory...

**Comment by adrienne** [comment]
I also have this issue, @engram-design

**Comment by evanwarner** [comment]
Ugh, sorry to keep the pressure on, but this issue is becoming a killer. One of my sites with a lot of fields takes 4+ _minutes_ to load a field edit page.

**Comment by engram-design** [comment]
I'm hesitant to roll my own solution, if Craft is going to address this in core, which I believe is on the cards.

**Comment by leevigraham** [comment]
For reference: https://github.com/craftcms/cms/issues/5792

**Comment by engram-design** [comment]
This is fixed in [2.5.0](https://github.com/verbb/super-table/releases/tag/2.5.0).

Please note you will also need [Craft 3.4.22](https://github.com/craftcms/cms/releases/tag/3.4.22)
